### DISCUSSION TITLE
ICE scratch removal using ImageMagick

### DISCUSSION THREAD

**Discussion by rleaver152** [ImageMagick/ImageMagick#2428] [Help]
Apologies if a FAQ,  but I have some RGBI tiff's  from colour negative scans made with ICE infrared dust/scratch detection  (Nikon Super Coolscan 4000).   The old negatives have long scratches on them which the IR parts of the scans isolate.   Is there any current work with ImageMagick that deals with automatic scratch removal using the infra red channel of a raw scan?    Or any pointers to any command line work in the area?   Happy to post an example if helpful.

Thanks
Richard

**Comment by fmw42** [comment]
Please do post an example

**Reply by rleaver152** [reply]
Fred many thanks - here's one.   Processed jpg and tiff  plus the original raw tiff with the IR info.

temporary Wetransfer uploads.

https://we.tl/t-RVcuV7UKVu
https://we.tl/t-feYcJ2HzNN
https://we.tl/t-lFTIKXhfNy
thanks,  best  Richard

**Reply by fmw42** [reply]
I am confused. Which defects do you want? I do not see any of the long vertical lines in the alpha channel (presumably your IR channel). I only see one small prominent defect. Can you explain in more detail and perhaps identify better what defects you want to find and then what you want to do with them. I note that if you threshold the alpha channel then that could be used with inpainting tools to fill in the scratches. Possibly snibgo's poisson blending or his hole filling tools could be used. But ImageMagick has neither of these in its current tool set. See http://im.snibgo.com

**Reply by rleaver152** [reply]
Fred - sorry - its those elongated blue lines running through the negative.  Let me see if I can find better examples.   Essentially anything that can be detected as long scratches.  B/W film cannot use IR (as the silver oxide is transparent) so a generic solution would be helpful.

**Reply by fmw42** [reply]
The lines are primarily in the blue channel and to a lesser extent in the green channel in those image. Not in the alpha or IR channel if they are the same.

**Reply by rleaver152** [reply]
Thanks - I'll try and find better examples.

**Comment by snibgo** [comment]
As Fred suggests, if you have a mask of where the scratches are, inpainting can be used. I would do it by making those pixels transparent, then "filling the holes".

If you have an example with a mask in the alpha channel, I can give it a try.
